scipy/spatial/transform/rotation.py

```diff
@@ -1460,3 +1462,54 @@ def __getitem__(self, indexer):
                [ 0.57735027,  0.57735027, -0.57735027,  0.        ]])
         """
         return self.__class__(self._quat[indexer], normalized=True)
+
+    @classmethod
+    def random(cls, num=None, random_state=None):
+        """Generate uniformly distributed rotations.
+
+        Parameters
+        ----------
+        num : None or int
+            Number of random rotations to generate. If None, then a single
+            rotation is generated. Default is None.
+        random_state : None, int or `numpy.random.RandomState` object
+            If RadomState object, used for generator state. If integer, used
+            to seed the random generator. If None, uses global `np.random`
+            random state. Default is None.
+
+        Returns
+        -------
+        random_rotation : `Rotation` instance
+            Contains a single rotation if `num` is None. Otherwise contains a
+            stack of `num` rotations.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Sample a single rotation:
+
+        >>> R.random().as_quat()
+        array([-0.07838344, -0.25840318, -0.24463331, -0.93125634])
+        >>> R.random(None).as_quat()
+        array([-0.24475285,  0.58126954, -0.54600616,  0.55145176])
+
+        Sample a stack of rotations:
+
+        >>> R.random(1).as_quat()
+        array([[ 0.43748455,  0.23734269, -0.80606684,  0.32020612]])
+        >>> R.random(5).as_quat()
+        array([[ 0.35598949,  0.62048719, -0.69368101, -0.08410586],
```

Comment:
What if we use Euler angles instead? To roughly demonstrate "uniformity".

Reply:
I'm not sure if that is an effective demonstration of uniformity, but still a better suggestion than quaternions.